inspirehep/modules/orcid/tasks.py

```diff
@@ -0,0 +1,168 @@
+    """ Extracts the authors with valid orcid credentials from the list of authors
+    of a given record in json format.
+    """
+    logger.info("Converted " + json.get('control_number'))
+    resolver = Resolver(pid_type='literature',
+                        object_type='rec', getter=lambda x: x)
+    record_id = resolver.resolve(json.get('control_number'))[
+        0].object_uuid
+    authors = get_orcid_valid_authors(json)
+    token = None
+    author_orcid = ''
+    authors_with_orcid_credentials = []
+    for author in authors:
+        try:
+            token, author_orcid = get_authors_credentials(author)
+        except AttributeError:
+            continue
+        try:
+            authors_with_orcid_credentials.append((InspireOrcidRecords.query.filter_by(
+                orcid=author_orcid, record_id=record_id).first().put_code, token, author_orcid, record_id))
+        except AttributeError:
+            authors_with_orcid_credentials.append(
+                ([], token, author_orcid, record_id))
+            continue
+    return authors_with_orcid_credentials
+
+
+@shared_task(ignore_result=True)
+def delete_from_orcid(sender, api=None):
+    """ Deletes a record from orcid.
+    """
+    if not api:
+        api = current_app.extensions['inspire-orcid'].orcid_api
+
+    resolver = Resolver(pid_type='literature',
+                        object_type='rec', getter=lambda x: x)
+    record_id = resolver.resolve(sender.get('control_number'))[
+        0].object_uuid
+    records = InspireOrcidRecords.query.filter_by(record_id=record_id).all()
+    for record in records:
+        api.remove_record(record.orcid, record.token, 'work', record.put_code)
+        with db.session.begin_nested():
+            db.session.delete(record)
+        db.session.commit()
+
+
+@shared_task(ignore_result=True)
+def send_to_orcid(sender, api=None):
+    """Sends records to orcid."""
+    index, doc_type = schema_to_index(sender['$schema'])
+    if doc_type == 'hep':
+        logger.info("Sending " + sender.get('control_number') + " to orcid.")
+        try:
+            if not api:
+                api = current_app.extensions['inspire-orcid'].orcid_api
+            orcid_json = convert_to_orcid(sender)
+            authors = prepare_authors_data_for_pushing_to_orcid(sender)
+            for author in authors:
+                try:
+                    put_code = author[0]
+                    token = author[1]
+                    author_orcid = author[2]
+                    record_id = author[3]
+                    if not put_code:
+                        put_code = api.add_record(  # try-continue
+                            author_orcid, token, 'work', orcid_json)
+                        orcid_log_record = InspireOrcidRecords(
+                            orcid=author_orcid,
+                            record_id=record_id,
+                            put_code=put_code,
+                            token=token)
```

Comment:
Why here you store the token? Isn't this de-normalizing the token information which is already in other tables?